lib/LWP/Protocol/http.pm

```diff
@@ -42,6 +42,8 @@ sub _new_socket
 	    $@ =~ /\b(Crypt-SSLeay can't verify hostnames)\b/
 	) {
 	    $status .= " ($1)";
+	} elsif ($@) {
+	    $status .= " ($@)";
 	}
 	die "$status\n\n$@";
```

Comment:
Is `$@` already displayed here?

Reply:
I am not sure why the $@ string is appended after the newlines at die(). I was displaying the error using the HTTP::Response->status_line() method which properly showed the additional "(Connection refused)" string after this change.

Before:
```
500 Can't connect to example.com:443
```
After:
```
500 Can't connect to example.com:443 (Connection refused)
```

Looking at the parse() function in my HTTP/Response.pm I see this:
```
 25     my($class, $str) = @_;
 26     my $status_line;
 27     if ($str =~ s/^(.*)\n//) {
 28         $status_line = $1;
 29     }
 30     else {
 31         $status_line = $str;
 32         $str = "";
 33     }
```
... so it looks to me like only the part leading up to the first newline is utilized when such a character is present.

Reply:
You may see the full error message in content, that is, $res->content.

Reply:
... after digging around some more with Carp cluck() I notice that I am not actually using parse(), but instead have as stack looking like this:
```
HTTP::Response::new()
LWP::UserAgent::_new_response()
LWP::UserAgent::send_request()
```

... and in LWP::UserAgent::send_request() the code does this:
```
 201                     my $full = $@;
 202                     (my $status = $@) =~ s/\n.*//s;
 203                     $status =~ s/ at .* line \d+.*//s;  # remove file/line number
 204                     my $code = ($status =~ s/^(\d\d\d)\s+//) ? $1 : &HTTP::Status::RC_INTERNAL_SERVER_ERROR;
 205                     $response = _new_response($request, $code, $status, $full);
```

So similarly to parse() it only keeps the part prior to the newline as the status string.